types/seeed/senseCAPT1000A/uplink.js

```diff
@@ -0,0 +1,568 @@
+      collectTime = getUTCTimestamp(dataValue.substring(8, 16));
+      measurement = {
+        sosEvent: eventList[6],
+        wifiScan: getMacAndRssiObj(dataValue.substring(16, 72)),
+        batteryLevel: getBattery(dataValue.substring(72, 74)),
+        timestamp: collectTime,
+      };
+      break;
+    case "0B":
+      eventList = getEventStatus(dataValue.substring(0, 6));
+      collectTime = getUTCTimestamp(dataValue.substring(8, 16));
+      measurement = {
+        sosEvent: eventList[6],
+        bleScan: getMacAndRssiObj(dataValue.substring(16, 58)),
+        batteryLevel: getBattery(dataValue.substring(58, 60)),
+        timestamp: collectTime,
+      };
+      break;
+    case "0D": {
+      const errorCode = getInt(dataValue);
+      let error = "";
+      switch (errorCode) {
+        case 0:
+          error = "GNSS_SCAN_TIME_OUT";
+          break;
+        case 1:
+          error = "WIFI_SCAN_TIME_OUT";
+          break;
+        case 2:
+          error = "WIFI_GNSS_SCAN_TIME_OUT";
+          break;
+        case 3:
+          error = "GNSS_WIFI_SCAN_TIME_OUT";
+          break;
+        case 4:
+          error = "BEACON_SCAN_TIME_OUT";
+          break;
+        case 5:
+          error = "BEACON_WIFI_SCAN_TIME_OUT";
+          break;
+        case 6:
+          error = "BEACON_GNSS_SCAN_TIME_OUT";
+          break;
+        case 7:
+          error = "BEACON_WIFI_GNSS_SCAN_TIME_OUT";
+          break;
+        case 8:
+          error = "FAILED_TO_OBTAIN_THE_UTC_TIMESTAMP";
+          break;
+        default:
+          break;
+      }
+      measurement = { errorCode, error };
+      break;
+    }
+    default:
+      break;
+  }
+  return measurement;
+}
+
+function messageAnalyzed(messageValue) {
+  try {
+    const frames = unpack(messageValue);
+    const measurementResultArray = [];
+    for (let i = 0; i < frames.length; i++) {
+      const item = frames[i];
+      const { dataId } = item;
+      const { dataValue } = item;
+      const measurementArray = deserialize(dataId, dataValue);
+      measurementResultArray.push(measurementArray);
+    }
+    return measurementResultArray;
+  } catch (e) {
+    return e.toString();
+  }
+}
+
+function getUpShortInfo(messageValue) {
+  return {
+    batteryLevel: getBattery(messageValue.substring(0, 2)),
+    firmwareVersion: getSoftVersion(messageValue.substring(2, 6)),
+    hardwareVersion: getHardVersion(messageValue.substring(6, 10)),
+    workMode: getWorkingMode(messageValue.substring(10, 12)),
+    heartbeatInterval: getOneWeekInterval(messageValue.substring(14, 18)),
+    periodicInterval: getOneWeekInterval(messageValue.substring(18, 22)),
+    eventInterval: getOneWeekInterval(messageValue.substring(22, 26)),
+    sosMode: getSOSMode(messageValue.substring(28, 30)),
+  };
+}
+
+function isEmpty(obj) {
+  return Object.keys(obj).length === 0;
+}
+
+function consume(event) {
+  const payload = event.data.payloadHex;
+  const result = messageAnalyzed(payload)[0];
```

Comment:
can there be more than one message in one uplink? if yes iterate

Reply:
No